src/App.tsx

```diff
@@ -1,14 +1,76 @@
-import React from 'react';
+import {
+  useMemo,
+  useState,
+  useCallback,
+  useEffect,
+} from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { DropdownMenu } from './DropdownMenu';
+import { Person } from './types/Person';
 
-export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+const debounce = <T extends string[]>(
+  f: (...args: T) => void,
+  delay: number,
+) => {
+  let timerId: NodeJS.Timeout;
+
+  return (...args: T) => {
+    clearTimeout(timerId);
+    timerId = setTimeout(f, delay, ...args);
+  };
+};
+
+export const App = () => {
+  const [query, setQuery] = useState<string>('');
+  const [appliedQuery, setAppliedQuery] = useState<string>('');
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [showSuggestions, setShowSuggestions] = useState(false);
+
+  useEffect(() => {
+    if (!query.trim()) {
+      setShowSuggestions(false);
+    }
+  }, [query]);
```

Comment:
can we replace this useEffect with a variable like?🤔

```jsx
const isSuggestionsVisible = !query.trim() && showSuggestions;
```

Reply:
Unfortunately, it does not work

Reply:
try this
`const isSuggestionsVisible = !!query.trim() && showSuggestions;`